### PR TITLE
Fix for artifact with thin-film materials and shadows

### PR DESCRIPTION
**Pull request by mrieggeramzn**
Smoothing out the ndl term does a decent enough job of fixing the visual artifact in most cases:

(Note that one material screenshot test will need updating after this PR)

ATOM-16840

### Before
before

### After
after

Signed-off-by: mrieggeramzn mriegger@amazon.com

### CODE REVIEW

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/BackLighting.azsli [hunk 1]
```diff
@@ -53,8 +63,10 @@ float3 GetBackLighting(Surface surface, LightingData lightingData, float3 lightI
             float litRatio = 1.0 - shadowRatio;
             if (litRatio)
             {
-                result = TransmissionKernel(surface.transmission.thickness * transmissionParams.w, rcp(transmissionParams.xyz)) * 
-                    saturate(dot(-surface.normal, dirToLight)) * lightIntensity * litRatio;
+                const float t = surface.transmission.thickness * transmissionParams.w;
```
Comment (Adi-Amazon) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/BackLighting.azsli:66]: super nit: more meaningful variable names - for example: t --> thickness

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/BackLighting.azsli [hunk 2]
```diff
@@ -23,6 +23,16 @@ float3 TransmissionKernel(float t, float3 s)
     return 0.25 * (1.0 / exp(exponent) + 3.0 / exp(exponent / 3.0));
 }
 
+float ThinObjectFalloff(const float3 l, const float3 n)
```
Comment (Adi-Amazon) [inline Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/BackLighting.azsli:26]: nit: meaningful variables names, example:
l --> lightDir
n --> surfaceNormal

**Review by Adi-Amazon — APPROVED** [review]
Please note my remarks, they are not a blocker but if you can change code accordingly, you have my full blessing :)

I'm guessing that in your example image you set the object's material to be marked as 'thin' to pin point the edge case? (I'm asking since it's not clear why it falls into the 'thin' category)

File: Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/BackLighting.azsli (hunk 2 above)
Comment (mrieggeramzn) [reply Gems/Atom/Feature/Common/Assets/ShaderLib/Atom/Features/PBR/BackLighting.azsli]: Agreed, changed.

**Review by dmcdiarmid-ly — APPROVED** [review]

**Comment by mrieggeramzn** [comment]
Yes that is correct. Basically there is a special (very green by default) thin-film material that I applied to the cylinder.

**Pull request merged**